src/frontend/src/btc/components/send/BtcSendTokenWizard.svelte

```diff
@@ -48,8 +58,75 @@
 	const close = () => dispatch('icClose');
 	const back = () => dispatch('icSendBack');
 
-	// TODO: implement send function when related services are ready
-	const send = () => {};
+	const send = async () => {
+		const network = nonNullish(networkId) ? mapNetworkIdToBitcoinNetwork(networkId) : undefined;
+
+		if (isNullish(network)) {
+			toastsError({
+				msg: { text: $i18n.send.error.no_btc_network_id }
+			});
+			return;
+		}
+
+		if (isNullishOrEmpty(destination)) {
+			toastsError({
+				msg: { text: $i18n.send.assertion.destination_address_invalid }
+			});
+			return;
+		}
+
+		if (invalidAmount(amount) || isNullish(amount)) {
+			toastsError({
+				msg: { text: $i18n.send.assertion.amount_invalid }
+			});
+			return;
+		}
+
+		if (isNullish(utxosFee)) {
+			toastsError({
+				msg: { text: $i18n.send.assertion.utxos_fee_missing }
+			});
+			return;
+		}
+
+		if (isNullish($sendToken)) {
+			toastsError({
+				msg: { text: $i18n.tokens.error.unexpected_undefined }
+			});
+			return;
+		}
+
+		if (isNullish($authIdentity)) {
+			toastsError({
+				msg: { text: $i18n.send.error.unexpected }
+			});
+			return;
+		}
+
+		dispatch('icNext');
+
+		try {
+			await sendBtc({
```

Comment:
just for posterity, may you please add a TODO for tracking analytics?

Reply:
Sure.